### PR TITLE
Feat(AUTH-CPC-740):  User Can Logout

### PR DESCRIPTION
**Pull request by PaulJ2001**
JIRA: link to jira ticket: 
https://champlainsaintlambert.atlassian.net/browse/CPC-740
## Context:
The logout functionality was incorrectly implemented, and it failed to remove the user account cookie upon logout.
## Changes
Implemented a functional logout feature that clears the user account session cookie and removes associated local storage keys (e.g., email, roles, username, UUID).
- Added logout methods in ApiGateway to facilitate logouts.
- Enhanced the Frontend logout function to include the removal of the Bearer account session cookie.
- Developed three comprehensive test cases to ensure the reliability of the new logout functionality.

### Before
Logout_Cookie
logout_UnremovedCookie

### After
Logout_Cookie
Logout_Success
Logout_CookieRemoved
logout_localStorage

## Dev notes (Optional)
None
## Linked pull requests (Optional)
None

### CODE REVIEW

File: api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/ApiGatewayControllerTest.java
Comment (SamuelNguyen2121) [inline api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/ApiGatewayControllerTest.java:1]: Tests look good despite that you were worried about not adding enough. In logout, i don't think there's not much errors you can get when doing it except for when the cookie is not cleared which you added a test for it

**Review by SamuelNguyen2121 — APPROVED** [review]
Very good!

File: api-gateway/src/main/resources/static/scripts/fragments/nav.html [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
+            window.location.href = '/#!/login';
+        }, function errorCallback(response) {
+            alert("Logout Error: Session may have expired. Redirecting to login page")
+            localStorage.removeItem("username")
+            localStorage.removeItem("email")
+            localStorage.removeItem("UUID")
+            localStorage.removeItem("roles")
+            window.location.href = '/#!/login';
+        });    }
 </script>
```
Comment (DylanBrass) [inline api-gateway/src/main/resources/static/scripts/fragments/nav.html:23]: Nice work handling the expired sessions, did you test it to make sure it works ?

**Review by DylanBrass — APPROVED** [review]

File: api-gateway/src/main/resources/static/scripts/fragments/nav.html (hunk 1 above)
Comment (PaulJ2001) [reply api-gateway/src/main/resources/static/scripts/fragments/nav.html:23]: Yes, I did do one test for an expired session in the ApiGatewayControllerTest:
SessionExpiredTest

**Pull request merged**